src/github.com/matrix-org/dendrite/mediaapi/writers/download.go

```diff
@@ -335,13 +497,18 @@ func (r *downloadRequest) fetchRemoteFileAndStoreMetadata(absBasePath types.Path
 		return &resErr
 	}
 
-	// TODO: generate thumbnails
+	go func() {
+		err := thumbnailer.GenerateThumbnails(finalPath, thumbnailSizes, r.MediaMetadata, activeThumbnailGeneration, db, r.Logger)
+		if err != nil {
+			r.Logger.WithError(err).Warn("Error generating thumbnails")
+		}
+	}()
```

Comment:
It's basically all covered here - pay attention to the control flow and hopefully it'll be clear: https://github.com/matrix-org/dendrite/blob/a1e62a2b0b9f76309a8aad96c5bd3fdeef8982f6/src/github.com/matrix-org/dendrite/mediaapi/writers/download.go#L292-L327